zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -40,89 +39,113 @@ def is_g1_not_zero(val):
+    # reclaimFunding allows the customer to withdraw their funds
+    # if the merchant has not funded their side of the channel yet.
     @sp.entry_point
     def reclaimFunding(self):
-        sp.verify(self.data.status == AWAITING_FUNDING)
-        sp.if self.data.custAddr == sp.sender:
-            sp.verify(self.data.custBal == self.data.custFunding)
-            sp.send(self.data.custAddr, self.data.custBal)
-            self.data.custBal = sp.tez(0)
-        sp.if self.data.merchAddr == sp.sender:
-            sp.verify(self.data.merchBal == self.data.merchFunding)
-            sp.send(self.data.merchAddr, self.data.merchBal)
-            self.data.merchBal = sp.tez(0)
+        # Verify that the channel is still AWAITING_MERCH_FUNDING.
+        sp.verify(self.data.status == AWAITING_MERCH_FUNDING)
+        # Only allow the customer to call the entrypoint.
+        sp.verify(self.data.custAddr == sp.sender)
+        # Verify that they had funded their side of the channel.
+        sp.verify(self.data.custBal == self.data.custFunding)
+        # Send the customer’s balance back to custAddr.
+        sp.send(self.data.custAddr, self.data.custBal)
+        # Set the channel status to FUNDING_RECLAIMED.
+        self.data.status = FUNDING_RECLAIMED
  
     # expiry can be called by the merchant to initiate channel closure.
     # The customer should call custClose using the latest state. Otherwise,
     # after the delay expires, the merchant will be able to claim all the
     # funds in the channel using merchClaim.
     @sp.entry_point
     def expiry(self):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is OPEN.
         sp.verify(self.data.status == OPEN)
+        # Record the time when the delay period will expire with self.data.delayExpiry.
+        # The time is calculated by adding the number of seconds in selfDelay to timestamp of
+        # the block it is confirmed in.
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
         self.data.status = EXPIRY
  
     # merchClaim can be called by the merchant if the customer has not called
     # custClose before the delay period has expired.
     @sp.entry_point
     def merchClaim(self):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is EXPIRY.
         sp.verify(self.data.status == EXPIRY)
+        # Verify that the delay period has passed.
         sp.verify(self.data.delayExpiry < sp.now)
+        # Send the total balance to the merchant.
         sp.send(self.data.merchAddr, self.data.custBal + self.data.merchBal)
-        self.data.custBal = sp.tez(0)
-        self.data.merchBal = sp.tez(0)
         self.data.status = CLOSED
 
     @sp.entry_point
     def custClose(self, custBal, merchBal, revLock, s1, s2):
```

Comment:
Add documentation.

Reply:
added:
```
    # custClose can be called by the customer to initiate a unilateral channel closure. It can only 
    # be called by the customer while the channel status is OPEN or EXPIRY. The entrypoint 
    # validates the merchant's blind signature over a tuple containing the channel ID, close flag, 
    # rev lock, customer closing balance and merchant closing balance.
```